### PR TITLE
add recurring and changes as per new endpoint updates

### PR DESCRIPTION
**Pull request by grepruby**

### CODE REVIEW

**Comment by grepruby** [comment]
Hello @Shopify/active-merchant @j-mutter @ntalbott 

I have added code for recurring and made changes as per the new endpoints, Please have a a look

File: lib/active_merchant/billing/gateways/pay_hub.rb [hunk 1]
```diff
@@ -48,149 +48,420 @@ class PayHubGateway < Gateway
...
       }
 
       def initialize(options={})
-        requires!(options, :orgid, :username, :password, :tid)
-
+        requires!(options, :organization_id, :oauth_token, :terminal_id, :version)
         super
       end
-
-      def authorize(amount, creditcard, options = {})
```
Comment (j-mutter) [inline lib/active_merchant/billing/gateways/pay_hub.rb]: Why was this removed? `authorize` is a standard gateway method and should probably still be implemented.

File: lib/active_merchant/billing/gateways/pay_hub.rb [hunk 2]
```diff
@@ -48,149 +48,420 @@ class PayHubGateway < Gateway
...
+        add_creditcard(post, creditcard, options)
         add_customer_data(post, options)
-
-        commit(post)
+        add_schedule(post, options)
+        
+        commit(:post, "recurring-bill", post, options)
       end
 
-      def refund(amount, trans_id, options={})
```
Comment (j-mutter) [inline lib/active_merchant/billing/gateways/pay_hub.rb]: Please don't change the method signature of the common gateway API methods

File: lib/active_merchant/billing/gateways/pay_hub.rb [hunk 3]
```diff
@@ -48,149 +48,420 @@ class PayHubGateway < Gateway
...
+          "Accept" => "application/json"
+        }
+        headers
+      end
+
+      def api_version(options)
+        options[:version] || @options[:version] || "v2"
+      end
+      
+      def commit(method, endpoint, post, options)
```
Comment (j-mutter) [inline lib/active_merchant/billing/gateways/pay_hub.rb]: You've added this `method` param, but never actually use it. Everything uses `ssl_post`

**Comment by j-mutter** [comment]
Also, the recurring functionality is deprecated and is being removed from Active Merchant.

File: lib/active_merchant/billing/gateways/pay_hub.rb (hunk 1 above)
Comment (grepruby) [inline lib/active_merchant/billing/gateways/pay_hub.rb]: @j-mutter

Yes correct, previously I was using `authorize`  but now asp er new changes we validates through oauth_token in headers.  You can see in [headers](https://github.com/grepruby/active_merchant/blob/payhub_updates/lib/active_merchant/billing/gateways/pay_hub.rb#L389) method
        

      def headers(options = {})
        oauth_token = options[:oauth_token] || @options[:oauth_token]

        headers = {
          "Authorization" => "Bearer " + oauth_token,
          "Content-Type" => "application/json",
          "Accept" => "application/json"
        }
        headers
      end

File: lib/active_merchant/billing/gateways/pay_hub.rb (hunk 2 above)
Comment (grepruby) [inline lib/active_merchant/billing/gateways/pay_hub.rb]: @j-mutter 

Actually we are no longer using `amount` in refund api, but we can add for maintaining signature, Please guide me.

**Comment by grepruby** [comment]
@j-mutter okay So should I remove `recurring` functionality from my code base, Please suggest?

**Comment by grepruby** [comment]
@j-mutter I was also wondering the CI(travis) didn't execute for my PR

**Comment by jasonwebster** [comment]
Closing this due to outstanding comments (can't change Active Merchant's contact), and staleness.
